packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -514,19 +514,20 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
    */
   private getDefaultEndpointPrefix(name: string, region: string) {
     const VPC_ENDPOINT_SERVICE_EXCEPTIONS: { [region: string]: string[] } = {
-      'cn-north-1': ['application-autoscaling', 'athena', 'autoscaling', 'awsconnector', 'cassandra',
```

Comment:
@pahud do you know why we are filtering for specific services? Is there ever a case where a service in  `cn` won't have the `cn` prefix?

Reply:
Hi @corymhall 

This is the list of all vpc interface endpoints in `cn-north-1`

```sh
% AWS_PROFILE=bjs AWS_REGION=cn-north-1 aws ec2 describe-vpc-endpoint-services | jq -r .ServiceNames
```
[
  "aws.sagemaker.cn-north-1.notebook",
  "aws.sagemaker.cn-north-1.studio",
  "cn.com.amazonaws.cn-north-1.application-autoscaling",
  "cn.com.amazonaws.cn-north-1.appmesh-envoy-management",
  "cn.com.amazonaws.cn-north-1.athena",
  "cn.com.amazonaws.cn-north-1.autoscaling",
  "cn.com.amazonaws.cn-north-1.awsconnector",
  "cn.com.amazonaws.cn-north-1.backup",
  "cn.com.amazonaws.cn-north-1.batch",
  "cn.com.amazonaws.cn-north-1.cassandra",
  "cn.com.amazonaws.cn-north-1.cloudcontrolapi",
  "cn.com.amazonaws.cn-north-1.cloudformation",
  "cn.com.amazonaws.cn-north-1.codedeploy-commands-secure",
  "cn.com.amazonaws.cn-north-1.databrew",
  "cn.com.amazonaws.cn-north-1.dms",
  "cn.com.amazonaws.cn-north-1.ebs",
  "cn.com.amazonaws.cn-north-1.ec2",
  "cn.com.amazonaws.cn-north-1.ecr.api",
  "cn.com.amazonaws.cn-north-1.ecr.dkr",
  "cn.com.amazonaws.cn-north-1.eks",
  "cn.com.amazonaws.cn-north-1.elasticache",
  "cn.com.amazonaws.cn-north-1.elasticbeanstalk",
  "cn.com.amazonaws.cn-north-1.elasticfilesystem",
  "cn.com.amazonaws.cn-north-1.elasticfilesystem-fips",
  "cn.com.amazonaws.cn-north-1.emr-containers",
  "cn.com.amazonaws.cn-north-1.execute-api",
  "cn.com.amazonaws.cn-north-1.fsx",
  "cn.com.amazonaws.cn-north-1.imagebuilder",
  "cn.com.amazonaws.cn-north-1.iot.data",
  "cn.com.amazonaws.cn-north-1.iotsitewise.api",
  "cn.com.amazonaws.cn-north-1.iotsitewise.data",
  "cn.com.amazonaws.cn-north-1.kinesis-streams",
  "cn.com.amazonaws.cn-north-1.lambda",
  "cn.com.amazonaws.cn-north-1.license-manager",
  "cn.com.amazonaws.cn-north-1.monitoring",
  "cn.com.amazonaws.cn-north-1.rds",
  "cn.com.amazonaws.cn-north-1.redshift",
  "cn.com.amazonaws.cn-north-1.redshift-data",
  "cn.com.amazonaws.cn-north-1.s3",
  "cn.com.amazonaws.cn-north-1.sagemaker.api",
  "cn.com.amazonaws.cn-north-1.sagemaker.featurestore-runtime",
  "cn.com.amazonaws.cn-north-1.sagemaker.runtime",
  "cn.com.amazonaws.cn-north-1.securityhub",
  "cn.com.amazonaws.cn-north-1.servicecatalog",
  "cn.com.amazonaws.cn-north-1.sms",
  "cn.com.amazonaws.cn-north-1.sqs",
  "cn.com.amazonaws.cn-north-1.states",
  "cn.com.amazonaws.cn-north-1.sts",
  "cn.com.amazonaws.cn-north-1.sync-states",
  "cn.com.amazonaws.cn-north-1.synthetics",
  "cn.com.amazonaws.cn-north-1.transcribe.cn",
  "cn.com.amazonaws.cn-north-1.transcribestreaming",
  "cn.com.amazonaws.cn-north-1.transfer",
  "cn.com.amazonaws.cn-north-1.xray",
  "com.amazonaws.cn-north-1.codebuild",
  "com.amazonaws.cn-north-1.codecommit",
  "com.amazonaws.cn-north-1.datasync",
  "com.amazonaws.cn-north-1.dynamodb",
  "com.amazonaws.cn-north-1.ec2messages",
  "com.amazonaws.cn-north-1.ecs",
  "com.amazonaws.cn-north-1.ecs-agent",
  "com.amazonaws.cn-north-1.ecs-telemetry",
  "com.amazonaws.cn-north-1.elasticbeanstalk-health",
  "com.amazonaws.cn-north-1.elasticloadbalancing",
  "com.amazonaws.cn-north-1.elasticmapreduce",
  "com.amazonaws.cn-north-1.git-codecommit",
  "com.amazonaws.cn-north-1.glue",
  "com.amazonaws.cn-north-1.kinesis-firehose",
  "com.amazonaws.cn-north-1.kms",
  "com.amazonaws.cn-north-1.logs",
  "com.amazonaws.cn-north-1.s3",
  "com.amazonaws.cn-north-1.secretsmanager",
  "com.amazonaws.cn-north-1.sns",
  "com.amazonaws.cn-north-1.ssm",
  "com.amazonaws.cn-north-1.ssmmessages",
  "com.amazonaws.cn-north-1.storagegateway",
  "com.amazonaws.cn-north-1.swf"
]

And in `cn-northwest-1`

```sh
% AWS_PROFILE=bjs AWS_REGION=cn-northwest-1 aws ec2 describe-vpc-endpoint-services | jq -r .ServiceNames
```
[
  "aws.sagemaker.cn-northwest-1.notebook",
  "aws.sagemaker.cn-northwest-1.studio",
  "cn.com.amazonaws.cn-northwest-1.account",
  "cn.com.amazonaws.cn-northwest-1.application-autoscaling",
  "cn.com.amazonaws.cn-northwest-1.appmesh-envoy-management",
  "cn.com.amazonaws.cn-northwest-1.athena",
  "cn.com.amazonaws.cn-northwest-1.autoscaling",
  "cn.com.amazonaws.cn-northwest-1.awsconnector",
  "cn.com.amazonaws.cn-northwest-1.backup",
  "cn.com.amazonaws.cn-northwest-1.batch",
  "cn.com.amazonaws.cn-northwest-1.cassandra",
  "cn.com.amazonaws.cn-northwest-1.cloudcontrolapi",
  "cn.com.amazonaws.cn-northwest-1.cloudformation",
  "cn.com.amazonaws.cn-northwest-1.codedeploy-commands-secure",
  "cn.com.amazonaws.cn-northwest-1.databrew",
  "cn.com.amazonaws.cn-northwest-1.dms",
  "cn.com.amazonaws.cn-northwest-1.ebs",
  "cn.com.amazonaws.cn-northwest-1.ec2",
  "cn.com.amazonaws.cn-northwest-1.ecr.api",
  "cn.com.amazonaws.cn-northwest-1.ecr.dkr",
  "cn.com.amazonaws.cn-northwest-1.eks",
  "cn.com.amazonaws.cn-northwest-1.elasticache",
  "cn.com.amazonaws.cn-northwest-1.elasticbeanstalk",
  "cn.com.amazonaws.cn-northwest-1.elasticfilesystem",
  "cn.com.amazonaws.cn-northwest-1.elasticfilesystem-fips",
  "cn.com.amazonaws.cn-northwest-1.emr-containers",
  "cn.com.amazonaws.cn-northwest-1.execute-api",
  "cn.com.amazonaws.cn-northwest-1.fsx",
  "cn.com.amazonaws.cn-northwest-1.imagebuilder",
  "cn.com.amazonaws.cn-northwest-1.iot.data",
  "cn.com.amazonaws.cn-northwest-1.kinesis-streams",
  "cn.com.amazonaws.cn-northwest-1.lambda",
  "cn.com.amazonaws.cn-northwest-1.license-manager",
  "cn.com.amazonaws.cn-northwest-1.monitoring",
  "cn.com.amazonaws.cn-northwest-1.polly",
  "cn.com.amazonaws.cn-northwest-1.rds",
  "cn.com.amazonaws.cn-northwest-1.redshift",
  "cn.com.amazonaws.cn-northwest-1.redshift-data",
  "cn.com.amazonaws.cn-northwest-1.s3",
  "cn.com.amazonaws.cn-northwest-1.sagemaker.api",
  "cn.com.amazonaws.cn-northwest-1.sagemaker.featurestore-runtime",
  "cn.com.amazonaws.cn-northwest-1.sagemaker.runtime",
  "cn.com.amazonaws.cn-northwest-1.securityhub",
  "cn.com.amazonaws.cn-northwest-1.servicecatalog",
  "cn.com.amazonaws.cn-northwest-1.sms",
  "cn.com.amazonaws.cn-northwest-1.sqs",
  "cn.com.amazonaws.cn-northwest-1.states",
  "cn.com.amazonaws.cn-northwest-1.sts",
  "cn.com.amazonaws.cn-northwest-1.sync-states",
  "cn.com.amazonaws.cn-northwest-1.synthetics",
  "cn.com.amazonaws.cn-northwest-1.transcribe.cn",
  "cn.com.amazonaws.cn-northwest-1.transcribestreaming",
  "cn.com.amazonaws.cn-northwest-1.transfer",
  "cn.com.amazonaws.cn-northwest-1.workspaces",
  "cn.com.amazonaws.cn-northwest-1.xray",
  "com.amazonaws.cn-northwest-1.codebuild",
  "com.amazonaws.cn-northwest-1.codecommit",
  "com.amazonaws.cn-northwest-1.datasync",
  "com.amazonaws.cn-northwest-1.dynamodb",
  "com.amazonaws.cn-northwest-1.ec2messages",
  "com.amazonaws.cn-northwest-1.ecs",
  "com.amazonaws.cn-northwest-1.ecs-agent",
  "com.amazonaws.cn-northwest-1.ecs-telemetry",
  "com.amazonaws.cn-northwest-1.elasticbeanstalk-health",
  "com.amazonaws.cn-northwest-1.elasticloadbalancing",
  "com.amazonaws.cn-northwest-1.elasticmapreduce",
  "com.amazonaws.cn-northwest-1.git-codecommit",
  "com.amazonaws.cn-northwest-1.glue",
  "com.amazonaws.cn-northwest-1.kinesis-firehose",
  "com.amazonaws.cn-northwest-1.kms",
  "com.amazonaws.cn-northwest-1.logs",
  "com.amazonaws.cn-northwest-1.s3",
  "com.amazonaws.cn-northwest-1.secretsmanager",
  "com.amazonaws.cn-northwest-1.sns",
  "com.amazonaws.cn-northwest-1.ssm",
  "com.amazonaws.cn-northwest-1.ssmmessages",
  "com.amazonaws.cn-northwest-1.storagegateway",
  "com.amazonaws.cn-northwest-1.swf"
]

I am guessing the old services are having `com.amazonaws` prefix while new services are coming with `cn.com.amazonaws` and sagemaker is an exception.

Maybe we should make `cn.com.amazonaws` as the default for CN regions and just list all services with `com.amazonaws` instead?

Reply:
hmmm I am afraid it's still very challenging to do that.

Some vpc endpoints are only available in `cn-northwest-1` and some only available in `cn-north-1`

only available in `cn-north-1`: `iotsitewise.api`, `iotsitewise.data`
only available in `cn-northwest-1`: `account`, `workspaces`